plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/InitialAuthConfig.java

```diff
@@ -32,52 +40,84 @@
+    @VisibleForTesting
+    protected static final String VALID_DOCKER_PROPERTY_NAME_EXAMPLE = CONFIG_PREFIX + "registry_name.parameter_name";
 
     /** For testing purposes */
     public InitialAuthConfig() {
     }
 
     @Inject
     public InitialAuthConfig(ConfigurationProperties configurationProperties) {
-        String username = null, password = null;
-        for (Map.Entry<String, String> e : configurationProperties.getProperties(CONFIGURATION_PREFIX_PATTERN).entrySet()) {
-            final String classifier = e.getKey().replaceFirst(CONFIGURATION_PREFIX, "");
-            switch (classifier) {
-                case "url": {
-                    serverAddress = e.getValue();
-                    break;
-                }
-                case "username": {
-                    username = e.getValue();
-                    break;
-                }
-                case "password": {
-                    password = e.getValue();
-                    break;
-                }
-            }
-        }
-        if (!isNullOrEmpty(serverAddress) && !isNullOrEmpty(username) && !isNullOrEmpty(password)) {
-            predefinedConfig = DtoFactory.newDto(AuthConfig.class).withUsername(username)
-                                                                  .withPassword(password);
+        Map<String, String> authProperties = configurationProperties.getProperties(CONFIGURATION_PREFIX_PATTERN);
+
+        Set<String> registryPrefixes = authProperties.entrySet()
+                                                     .stream()
+                                                     .map(property -> getRegistryPrefix(property.getKey()))
+                                                     .collect(Collectors.toSet());
+
+        Map<String, AuthConfig> configMap = new HashMap<>();
+        for (String regPrefix: registryPrefixes) {
+            String url = getPropertyValue(authProperties, regPrefix + URL);
+            String userName = getPropertyValue(authProperties, regPrefix + USER_NAME);
+            String password = getPropertyValue(authProperties, regPrefix + PASSWORD);
+
+            configMap.put(url, newDto(AuthConfig.class).withUsername(userName).withPassword(password));
         }
+
+        authConfigs = newDto(AuthConfigs.class).withConfigs(configMap);
     }
 
+    /**
+     * Returns docker model config file {@link AuthConfig}
+     */
     public AuthConfigs getAuthConfigs() {
-        AuthConfigs authConfigs = DtoFactory.newDto(AuthConfigs.class);
-        if (predefinedConfig != null) {
-            authConfigs.getConfigs().put(serverAddress, predefinedConfig);
-        }
         return authConfigs;
     }
 
+    private String getRegistryPrefix(String propertyName) {
+        String[] parts = propertyName.replaceFirst(CONFIG_PREFIX, "").split("\\.");
+
+        if (parts.length < 2) {
+            throw new IllegalArgumentException(format("You missed '.' in property '%s'. Valid credential registry format is '%s'",
```

Comment:
I'm not sure of the "You missed" style?
User seeing that error may not be the one that configured the property ?